src/ServerRequestFactory.php

```diff
@@ -70,6 +71,7 @@ public static function fromGlobals(
         );
 
         return $request
+            ->withProtocolVersion(static::marshalProtocolVersion($_SERVER))
             ->withCookieParams($cookies ?: $_COOKIE)
```

Comment:
I'm little worried about this. Why we need to create 5 instances of `ServerRequest`? `ServerRequest` should be able to be constructed with all possible args. Maybe we can named constructor here?